### PR TITLE
Fix/etk fse universal theme clean up

### PR DESCRIPTION
**Pull request by agrullon95**
#### Proposed Changes

Clean up for ETK Full site editing activation code.
Removed unused/unnecessary code related to activating FSE and kept code to hide "Menus" submenu in Calypso.

The logic for hiding the `Menus` submenu was kept to ensure consistency due to the fact that it was hidden in the FSE beta.

This should be merged after the D84305-code is merged.

#### Testing Instructions

* Activate FSE enabled theme (Blank Canvas). You should not see the `Menus` submenu under the appearance tab.
* Activate a non-FSE enabled theme (Dara). You should see the `Menus` submenu under the appearance tab.
* The AMP menu should still appear in calypso/wp-admin side menu if plugin is activated.
* Ensure site-editor is still loading correctly.

#### Pre-merge Checklist

Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in [Simple](P9HQHe-k8-p2-p2), [Atomic](P9HQHe-jW-p2-p2), and [self-hosted Jetpack sites](PCYsg-g6b-p2-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Are we memoizing when appropriate (for expensive computations)? More info in [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md) and [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors)
- [ ] Have we sent any new strings [for translation](PCYsg-1vr-p2-p2) ASAP?

Related to #61107

### CODE REVIEW

File: apps/editing-toolkit/editing-toolkit-plugin/wpcom-universal-themes/index.php [hunk 1]
```diff
@@ -9,168 +9,6 @@
...
-}
-
-/**
- * Unhooks the Gutenberg notice for Full Site Editing.
- *
- * @return void
- */
-function unload_core_fse() {
-	remove_action( 'admin_menu', __NAMESPACE__ . '\hide_nav_menus_submenu' );
-	remove_action( 'admin_notices', 'gutenberg_full_site_editing_notice' );
```
Comment (Addison-Stavlo) [inline apps/editing-toolkit/editing-toolkit-plugin/wpcom-universal-themes/index.php:107]: do we still need this bit?

Comment (agrullon95) [reply apps/editing-toolkit/editing-toolkit-plugin/wpcom-universal-themes/index.php:107]: No, we don't need this. `gutenberg_full_site_editing_notice` is not being used or referenced in other files.

I will still use the `hide_nav_menus_submenu` function to ensure we are still hiding the "Menus" sub menu for FSE themes.

**Comment by Addison-Stavlo** [comment]
noting the changes here require D84305-code to be applied first.

**Review by Addison-Stavlo — APPROVED** [review]
> Activate FSE enabled theme (Blank Canvas). You should not see the Menus submenu under the appearance tab.
Activate a non-FSE enabled theme (Dara). You should see the Menus submenu under the appearance tab.
The AMP menu should still appear in calypso/wp-admin side menu if plugin is activated.
Ensure site-editor is still loading correctly.

✅  - that all tests well!

**Review by creativecoder — APPROVED** [review]
Tested on a Simple site: Site editor loads without PHP errors for FSE themes, Customizer loads fine for FSE and legacy themes.

**Pull request merged**